### PR TITLE
Interpolation rs

### LINKED ISSUE #2178: Interpolation Search
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [ ] Dart
- [x] Go
- [x] Java
- [x] JS
- [ ] Kotlin
- [x] Php
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by SubCoder1**
Fixes https://github.com/jainaman224/Algo_Ds_Notes/issues/2178

#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Input is taken dynamically.
- [X] Sample Input / Output is added at the end of file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added implementation of Interpolation Search in Rust.

#### Languages Used:
- Rust

#### Files Added:
- Interpolation_Search.rs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Interpolation_Search/Interpolation_Search.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        Err(hi + 1)
+    } else if *target < arr[lo] {
+        Err(lo)
+    } else {
+        Ok(interpolant)
+    }
+}
+
+fn main() {
+    println!("{:?}", interpolation_search(&[1,2,3,4,6], &1));
```
Comment (jainaman224) [inline Interpolation_Search/Interpolation_Search.rs]: Take input from user.

File: Interpolation_Search/Interpolation_Search.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}
```
Comment (jainaman224) [inline Interpolation_Search/Interpolation_Search.rs]: Add sample input and output.

File: Linear_Search/Linear_Search.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+fn linear_search<T: Ord>(vec: &Vec<T>, key: &T) -> Option<usize> {
```
Comment (jainaman224) [inline Linear_Search/Linear_Search.rs]: This file should not be a part of this pr.

**Review by jainaman224 — CHANGES_REQUESTED** [review]

**Comment by somya-kapoor** [comment]
@SubCoder1 Are you working on it?

File: Linear_Search/Linear_Search.rs (hunk 3 above)
Comment (SubCoder1) [reply Linear_Search/Linear_Search.rs]: Resolved this issue & updated my PR branch.

**Comment by SubCoder1** [comment]
@jainaman224 I've updated my PR with required changes.

File: Interpolation_Search/Interpolation_Search.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
+pub fn interpolation_search(arr: &[i32], target: &i32) -> Result<usize, usize> {
+    // Handle empty sequence.
+    if arr.is_empty() {
+        return Err(0);
+    }
+
+    // Setup variable storing iteration informaion.
```
Comment (singh-shreya6) [inline Interpolation_Search/Interpolation_Search.rs:7]: Please change these single line comments to multi line comment.

File: Interpolation_Search/Interpolation_Search.rs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    let mut hi = arr.len() - 1;
+    let mut lo = 0_usize;
+    let mut interpolant = 0_usize;
+
+    // Main loop to calculate the interpolant.
+    loop {
+        let lo_val = arr[lo];
+        let hi_val = arr[hi];
+
+        // Three condition to exit the loop
```
Comment (singh-shreya6) [inline Interpolation_Search/Interpolation_Search.rs:20]: Please change these single line comments to multi line comment.

File: Interpolation_Search/Interpolation_Search.rs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        let my_int: i32 = input.trim().parse().expect("Not a valid number");
+        some_vec.push(my_int);
+    }
+    println!("Enter the element to be searched in the array :- ");    
+    let mut input = String::new();
+    io::stdin().read_line(&mut input).expect("Error reading input");
+    let my_int: i32 = input.trim().parse().expect("Not a valid number");
+
+    println!("{:?}", interpolation_search(&some_vec, &my_int));
+    /* Sample Input :-
```
Comment (singh-shreya6) [inline Interpolation_Search/Interpolation_Search.rs:71]: Please shift the sample input and output after closing bracket.

File: Interpolation_Search/Interpolation_Search.rs [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        1
+        2
+        3
+        4
+        6
+        Sample Output :-
+        The element is present at index: 4
+    */
+}
```
Comment (singh-shreya6) [inline Interpolation_Search/Interpolation_Search.rs:80]: This should be closed before i/o

**Review by singh-shreya6 — CHANGES_REQUESTED** [review]
Please make the changes and squash your commits by tomorrow to be considered in GSSoC.

**Pull request closed without merging**